app/routes/_index/route.tsx

```diff
@@ -1,19 +1,19 @@
 import { LinksFunction, LoaderFunctionArgs } from '@remix-run/node';
 import { Link, MetaFunction, useLoaderData, useNavigate, json } from '@remix-run/react';
-import { getEcomApi } from '~/api/ecom-api';
-import { HeroImage } from '~/components/hero-image/hero-image';
-import { ProductCard } from '~/components/product-card/product-card';
-import { ROUTES } from '~/router/config';
-import { getUrlOriginWithPath, isOutOfStock } from '~/utils';
+import { initializeEcomApi } from '~/lib/ecom/session';
+import { HeroImage } from '~/src/components/hero-image/hero-image';
+import { ProductCard } from '~/src/components/product-card/product-card';
+import { isOutOfStock } from '~/lib/utils';
 import styles from './index.module.scss';
 
 export const loader = async ({ request }: LoaderFunctionArgs) => {
-    const productsResponse = await getEcomApi().getPromotedProducts();
+    const ecomApi = await initializeEcomApi(request);
+    const productsResponse = await ecomApi.getPromotedProducts();
     if (productsResponse.status === 'failure') {
         throw json(productsResponse.error);
     }
 
-    return { products: productsResponse.body, canonicalUrl: getUrlOriginWithPath(request.url) };
```

Comment:
shouldn't we have this canonical url now?

Reply:
we should.. adding